core/shared/src/main/scala/org/http4s/EntityEncoder.scala

```diff
@@ -150,13 +148,13 @@ object EntityEncoder {
     stringEncoder.contramap(new String(_))
 
   implicit val chunkEncoder: EntityEncoder.Pure[Chunk[Byte]] =
-    simple(`Content-Type`(MediaType.application.`octet-stream`))(identity)
+    byteVectorEncoder.contramap(_.toByteVector)
 
   implicit val byteArrayEncoder: EntityEncoder.Pure[Array[Byte]] =
-    chunkEncoder.contramap(Chunk.array[Byte])
+    byteVectorEncoder.contramap(ByteVector.view)
 
   implicit def byteVectorEncoder[F[_]]: EntityEncoder[F, ByteVector] =
```

Comment:
Why is this `EntityEncoder[F, X]` while the others  are `EntityEncoder.Pure[X]`?

Reply:
Good catch, probably an oversight